vagabond/gui/AddEntity.cpp

```diff
@@ -281,26 +283,38 @@ void AddEntity::searchPdb()
 
 void AddEntity::loadConfSpaceView(std::string suffix)
 {
-	ConfSpaceView *view = new ConfSpaceView(this, &_obj);
-	
-	if (suffix == "refined_torsions")
-	{
-		view->setMode(rope::ConfTorsions);
-	}
-	else if (suffix == "atom_positions")
-	{
-		view->setMode(rope::ConfPositional);
-	}
-
-	try
-	{
-		view->show();
-	}
-	catch (const std::runtime_error &err)
-	{
-		BadChoice *bc = new BadChoice(this, err.what());
-		setModal(bc);
-	}
+        try
+        {
+                if (_obj.modelCount()==0)
+                {
+                        throw std::invalid_argument("No models detected, have you modeled your entities?");
+                }
+
+                ConfSpaceView *view = new ConfSpaceView(this, &_obj);
+                if (suffix == "refined_torsions")
+                {
+                        view->setMode(rope::ConfTorsions);
+                }
+                else if (suffix == "atom_positions")
+                {
+                        view->setMode(rope::ConfPositional);
+                }
+                        try
+                {
+                        view->show();
+                }
+                catch (const std::runtime_error &err)
+                {
+                        BadChoice *bc = new BadChoice(this, err.what());
+                        setModal(bc);
+                }
+        }
+        catch(std::invalid_argument &err)
```

Comment:
I would make this catch an `std::exception`. At the moment this catch statement is also used to catch an error in `Entity::makeTorsionDataGroup` which produces a `std::runtime_error`, which is no longer caught by this statement. Both `std::invalid_argument` and `std::runtime_error` are subclasses of `std::exception`, so you can handle them both in the same way.